### PR TITLE
Push notifs - Add Pushover support

### PR DESCRIPTION
**Pull request by LukeGotWood**
- Updated the config file so config options are organised into sections,

- Added [Pushover](https://pushover.net/) notification support,

- Updated the read me to reflect changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,22 @@
...
 
 The generated key will be in the format `o.Dorf43jdDIepfKeroPewfjeIUHJ4MrOP`.
 
-Open your client and paste the key into the `Pushbullet` config field of `Push Notifications` to enable phone notifications.
\ No newline at end of file
+Open your client and paste the key into the `Pushbullet` config field of `Push Notifications` to enable phone notifications.
+## Pushover
+You need to provide both a pushover user key and api token to use this service.
+
+Go [here](https://www.pushbullet.com/), login and your user key will be in the top right of the page.
```
Comment (dekvall) [inline README.md]: This should be pushover

File: src/main/java/dekvall/pushnotification/PushNotificationsPlugin.java [hunk 2]
```diff
@@ -52,37 +52,74 @@ PushNotificationsConfig provideConfig(ConfigManager configManager)
...
 			return;
 		}
 
 		HttpUrl url = new HttpUrl.Builder()
-			.scheme("https")
-			.host("api.pushbullet.com")
-			.addPathSegment("v2")
-			.addPathSegment("pushes")
-			.build();
+				.scheme("https")
```
Comment (dekvall) [inline src/main/java/dekvall/pushnotification/PushNotificationsPlugin.java]: Can you set your continuation indent to 4 so these don't get moved?

File: src/main/java/dekvall/pushnotification/PushNotificationsConfig.java [hunk 3]
```diff
@@ -3,14 +3,48 @@
 import net.runelite.client.config.Config;
 import net.runelite.client.config.ConfigGroup;
 import net.runelite.client.config.ConfigItem;
+import net.runelite.client.config.ConfigSection;
 
-@ConfigGroup("pushnotifications")
+@ConfigGroup(PushNotificationsConfig.GROUP)
```
Comment (dekvall) [inline src/main/java/dekvall/pushnotification/PushNotificationsConfig.java]: You can just keep this as is, i don't see it being used anywhere else

File: src/main/java/dekvall/pushnotification/PushNotificationsPlugin.java [hunk 4]
```diff
@@ -52,37 +52,74 @@ PushNotificationsConfig provideConfig(ConfigManager configManager)
...
+				.post(push)
+				.url(url)
+				.build();
 
 		sendRequest("Pushbullet", request);
 	}
 
+	private void handlePushover(NotificationFired event)
+	{
+		if(Strings.isNullOrEmpty(config.pushover_api()) && Strings.isNullOrEmpty(config.pushover_user()))
```
Comment (dekvall) [inline src/main/java/dekvall/pushnotification/PushNotificationsPlugin.java]: Shouldn't this be `||`?

**Review by dekvall — CHANGES_REQUESTED** [review]
Thanks! Also are you doing Hacktoberfest. Because if you are I think i need to opt-in the repo properly.

File: README.md (hunk 1 above)
Comment (LukeGotWood) [reply README.md]: Must have forgotten to change it in the copy and paste good spot!

File: src/main/java/dekvall/pushnotification/PushNotificationsConfig.java (hunk 3 above)
Comment (LukeGotWood) [reply src/main/java/dekvall/pushnotification/PushNotificationsConfig.java]: Fair point, I just saw it done this way on the Runelite Plugin wiki page so mimicked it

File: src/main/java/dekvall/pushnotification/PushNotificationsPlugin.java (hunk 4 above)
Comment (LukeGotWood) [reply src/main/java/dekvall/pushnotification/PushNotificationsPlugin.java]: Missed that in testing, Cheers!

**Comment by LukeGotWood** [comment]
I was looking to participate but I don't think I'll get all four PRs before the end of October so it's nothing to worry about. Honestly, I added the feature because its something I want for afking nightmare zone. Saved me making a whole new plugin plus I got to make my first PR which is very cool!

File: src/main/java/dekvall/pushnotification/PushNotificationsConfig.java [hunk 5]
```diff
@@ -3,14 +3,46 @@
...
+	)
+	String pushbulletSection = "pushbullet";
+
 	@ConfigItem(
-		keyName = "pushbullet",
-		name = "Pushbullet token",
-		description = "API token for pushbullet"
+			keyName = "pushbullet",
+			name = "Pushbullet token",
+			description = "API token for pushbullet",
```
Comment (dekvall) [inline src/main/java/dekvall/pushnotification/PushNotificationsConfig.java]: Can you dedent these new config fields as well?

File: src/main/java/dekvall/pushnotification/PushNotificationsPlugin.java [hunk 6]
```diff
@@ -22,7 +22,7 @@
 
 @Slf4j
 @PluginDescriptor(
-	name = "Push Notifications"
+		name = "Push Notifications"
```
Comment (dekvall) [inline src/main/java/dekvall/pushnotification/PushNotificationsPlugin.java]: And dedent this final one as well please :)

Comment (LukeGotWood) [reply src/main/java/dekvall/pushnotification/PushNotificationsPlugin.java]: Wow, I really don't have a good eye for spotting these things :)

**Review by dekvall — APPROVED** [review]

**Pull request merged**